examples/metalk8s-solution-example/ui/src/services/utils.js

```diff
@@ -30,3 +31,40 @@ export const useRefreshEffect = (refreshAction, stopRefreshAction) => {
     };
   }, [dispatch, refreshAction, stopRefreshAction]);
 };
+
+/* Check if a version is supported by comparaing it to the 'environmentVersion'
+   Here we decide that all the  "preminor of 1" versions are allowed
+   Example :
+   const versions = [
+              { version: '1.1.4-prod' },
+              { version: '1.2.0' },
+              { version: '0.1.5' },
+              { version: '1.0.5' },
+              { version: '1.0.2-dev' }];
+
+    Results: versions.filter(isVersionSupported('1.1.4')) === [
+              { version: '1.1.4-prod' },
+              { version: '1.0.5' },
+              { version: '1.0.2-dev' }
+
+    ]
+*/
+export const isVersionSupported = environmentVersion => item => {
+  if (environmentVersion) {
+    const environmentVersionMajor = semver.major(environmentVersion);
+    const environmentVersionMinor = semver.minor(environmentVersion);
+    const environmentVersionPatch = semver.patch(environmentVersion);
+
+    const parsedEnvironmentVersion = `${environmentVersionMajor}.${environmentVersionMinor}.${environmentVersionPatch}`;
+    const minimunSupportedVersion = `${environmentVersionMajor}.${
+      environmentVersionMinor > 1 ? environmentVersionMinor - 1 : 0
+    }.0`;
+
+    return (
+      semver.gte(item.version, minimunSupportedVersion) &&
+      semver.lte(item.version, parsedEnvironmentVersion)
+    );
+  } else {
+    return true;
+  }
```

Comment:
I'd say we can come up with a less verbose implementation, using `semver.parse`. Here's what I would imagine:

```javascript
export const isVersionSupported = envVersionStr => item => {
  let envVersion = semver.parse(envVersionStr);
  envVersion.prerelease = [];  // Reset pre-release info
  if (envVersionObj) {
    const minVersion = `${envVersion.major}.${Math.max(envVersion.minor - 1, 0)}`
    const versionRange = `${minVersion} - ${envVersion.format()}`
    return semver.statisfies(item.version, versionRange);
  } else {
    return true;
  }
}
```